### PR TITLE
Add email alerts on application server errors

### PR DESCRIPTION
**Pull request by suecarmol**
## Description
Added a setting in the production and staging environments that allow sending an email whenever there is an application error.

## Rationale
[//]: # (Why is this change required? What problem does it solve?)
This will help track any errors that may occur in the application.

## Phabricator Ticket
[//]: # (Link to the Phabricator ticket)
[T258144](https://phabricator.wikimedia.org/T258144)

## How Has This Been Tested?
[//]: # (- Did you add tests to your changes? Did you modify tests to accommodate your changes?)
[//]: # (- Can this change be tested manually? How?)
TBD

## Screenshots of your changes (if appropriate):
[//]: # (It can also be a GIF to prove that your changes are working)
N/A

## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: extlinks/settings/production.py [hunk 1]
```diff
@@ -2,15 +2,16 @@
...
-]
+CRON_CLASSES += ["extlinks.links.cron.TotalLinksCron"]
 
 # Redirect HTTP to HTTPS
 # SECURE_PROXY_SSL_HEADER is required because we're behind a proxy
 SECURE_SSL_REDIRECT = True
-SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
+SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")
+
+SERVER_EMAIL = "Wikipedia Library Card Platform <noreply@wikipedialibrary.wmflabs.org>"
```
Comment (Samwalton9) [inline extlinks/settings/production.py:16]: This (and in staging.py) should probably be `Wikilink` instead of `Wikipedia Library Card Platform`?

Comment (suecarmol) [reply extlinks/settings/production.py]: Thanks for the catch! Changed it!

**Comment by jsnshrmn** [comment]
These images were already being published separately:
https://hub.docker.com/r/wikipedialibrary/externallinks/tags
https://hub.docker.com/r/wikipedialibrary/eventstream/tags

**Comment by jsnshrmn** [comment]
okay, I've reworked our docker-related variable names and verified that this is pushing the correct images to dockerhub when I overwrite staging with it.
I did
```
docker system prune --all --force 
docker pull wikipedialibrary/eventstream:branch_staging
docker pull wikipedialibrary/externallinks:branch_staging
docker-compose up
```
and was able to get a working deployment without building anything.

I did find that we currently can have problems if we have workflow overruns between commits. I spun my wheels for a while trying to get an environment variable injected into a gh artifact name. I don't have this solved, but basically, we should be able to add a commit sha or action run id into the artifact name. It just doesn't seem to be working as I'd expect based on the documentation.

https://github.com/actions/upload-artifact#environment-variables-and-tilde-expansion
https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#env-context

**Comment by suecarmol** [comment]
I think I managed to inject a `run_id` into the artifact names. Is this what you meant? Let me know if I need to make any additional changes.

**Comment by jsnshrmn** [comment]
Thanks! That's exactly what I was hoping to accomplish! Maybe the env context is only available to tasks running in a shell? This looks good from the github actions perspective!

**Comment by jsnshrmn** [comment]
Fat-fingered the close button!

**Comment by suecarmol** [comment]
Before merging this to `master`, I would like to test this out on the staging environment. Otherwise, I think this PR is ready to merge

**Comment by jsnshrmn** [comment]
Okay, I deployed a new staging environment with the current staging branch. I updated the project wiki notes as well.
https://github.com/WikipediaLibrary/externallinks/wiki/Debian-Server-Setup
Note that this will not currently pull new images automatically, as we talked about adding backups first.

File: extlinks/settings/local.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
 from .base import *
 
 DEBUG = True
+
+DEFAULT_FROM_EMAIL = "<twlight.local@localhost.localdomain>"
```
Comment (jsnshrmn) [inline extlinks/settings/local.py:5]:
```suggestion
SERVER_EMAIL = "Wikilink Local <wikilink@localhost.localdomain>"
DEFAULT_FROM_EMAIL = SERVER_EMAIL
```

File: extlinks/settings/production.py [hunk 3]
```diff
@@ -12,3 +12,6 @@
 # SECURE_PROXY_SSL_HEADER is required because we're behind a proxy
 SECURE_SSL_REDIRECT = True
 SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")
+
+SERVER_EMAIL = "Wikilink <noreply@wikipedialibrary.wmflabs.org>"
```
Comment (jsnshrmn) [inline extlinks/settings/production.py:16]:
```suggestion
SERVER_EMAIL = "Wikilink <noreply@wikilink.wmflabs.org>"
```

File: extlinks/settings/staging.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from .base import *
+
+DEBUG = False
+
+ALLOWED_HOSTS += ["wikilink-staging.wmflabs.org"]
```
Comment (jsnshrmn) [inline extlinks/settings/staging.py:5]: We should have as few allowed hosts as possible.
```suggestion
ALLOWED_HOSTS = ["wikilink-staging.wmflabs.org"]
```

**Review by jsnshrmn — CHANGES_REQUESTED** [review]
Should use project and environment specific email addresses. See inline suggestions.

File: extlinks/settings/local.py (hunk 2 above)
Comment (suecarmol) [reply extlinks/settings/local.py]: Done!

File: extlinks/settings/production.py (hunk 3 above)
Comment (suecarmol) [reply extlinks/settings/production.py]: Done!

File: extlinks/settings/staging.py (hunk 4 above)
Comment (suecarmol) [reply extlinks/settings/staging.py]: That makes sense!

**Review by jsnshrmn — APPROVED** [review]

File: extlinks/settings/staging.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+# We can only run this in production, because we're connecting to the database
+# replicas internally.
+CRON_CLASSES += ["extlinks.links.cron.TotalLinksCron"]
+
+# Redirect HTTP to HTTPS
+# SECURE_PROXY_SSL_HEADER is required because we're behind a proxy
+SECURE_SSL_REDIRECT = True
+SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")
+
+SERVER_EMAIL = "Wikilink <noreply@wikilink.wmflabs.org>"
```
Comment (suecarmol) [inline extlinks/settings/staging.py:16]: I was thinking, should this be `wikilink-staging.wmflabs.org`?

**Review by jsnshrmn — APPROVED** [review]
This is looking great! Thanks for your work!

**Pull request merged**